controller/main.go

```diff
@@ -11,16 +11,46 @@ import (
 )
 
 func main() {
+	testVolumeArg := flag.String(
+		"test-volume",
+		"",
+		"The name of the volume that will have been created by the initializer and mounted on this controller, which can also be mounted on other nodes in the network",
+	)
+
+	testVolumeMountpointArg := flag.String(
+		"test-volume-mountpoint",
+		"",
+		"The filepath in the test controller's filesystem where the test volume will have been mounted by the initializer",
+	)
+
 	testNameArg := flag.String(
 		"test",
 		"",
 		"Comma-separated list of specific tests to run (leave empty or omit to run all tests)",
 	)
 
-	networkInfoFilepathArg := flag.String(
-		"network-info-filepath",
+	testImageNameArg := flag.String(
+		"test-image-name",
 		"",
-		"Filepath of file containing JSON-serialized representation of the network of service Docker containers",
+		"Name of Docker image of the service being tested",
+	)
+
+	subnetMaskArg := flag.String(
+		"subnet-mask",
+		"",
+		"Subnet mask of the Docker network that the test controller is running in",
+	)
+
+	testControllerIpArg := flag.String(
+		"test-controller-ip",
+		"",
+		"IP address of the Docker container running this test controller",
+	)
+
+	gatewayIpArg := flag.String(
```

Comment:
do we have to expose this to the user? it seems like we should choose a sensible default and enforce a range that they can use for IPs, feels like exposing an unnecessary implementation detail

Reply:
I think yes, since I model the initializer -> controller as more of a function call than something the user would call manually (e.g. the user would never actually run their own test controller image independent of the initializer - there'd be too much work).

Put another way though, what gateway IP from a subnet is chosen by the initializer is an implementation detail of the initializer; the test controller shouldn't need to know anything about how the initializer chooses it from a subnet range. The controller just needs to know what IPs are already taken and not use those.